components/core/tests/test-NetworkReader.cpp

```diff
@@ -233,17 +254,27 @@ TEST_CASE("network_reader_with_illegal_http_header_kv_pairs", "[NetworkReader]")
             // The CRLF-terminated headers should be rejected.
             std::unordered_map<std::string, std::string>{{"Legal-Name", "CRLF\r\n"}}
     );
-    clp::NetworkReader reader{
-            "https://httpbin.org/headers",
-            0,
-            false,
-            clp::CurlDownloadHandler::cDefaultOverallTimeout,
-            clp::CurlDownloadHandler::cDefaultConnectionTimeout,
-            clp::NetworkReader::cDefaultBufferPoolSize,
-            clp::NetworkReader::cDefaultBufferSize,
-            illegal_header_kv_pairs
-    };
-    auto const content = get_content(reader);
-    REQUIRE(content.empty());
-    REQUIRE(assert_curl_error_code(CURLE_BAD_FUNCTION_ARGUMENT, reader));
+    bool is_pass{false};
+    for (size_t i{0}; i < cNumMaxTrials; ++i) {
+        clp::NetworkReader reader{
+                "https://httpbin.org/headers",
+                0,
+                false,
+                clp::CurlDownloadHandler::cDefaultOverallTimeout,
+                clp::CurlDownloadHandler::cDefaultConnectionTimeout,
+                clp::NetworkReader::cDefaultBufferPoolSize,
+                clp::NetworkReader::cDefaultBufferSize,
+                illegal_header_kv_pairs
+        };
+        auto const content = get_content(reader);
+        if (false == content.empty()) {
+            continue;
+        }
```

Comment:
We don't need to retry for this unit test, right?
The header validation happens before sending HTTP requests. This unit test should always work regardless of server-end failures.

Reply:
yea, right, will get rid of it